File: shared/src/main/java/ru/privatenull/pnauth/security/HashAlgorithm.java
package ru.privatenull.pnauth.security;

public enum HashAlgorithm {
    PBKDF2,
    BCRYPT,
    ARGON2;

    public static HashAlgorithm parse(String value) {
        if (value == null || value.isBlank()) {
            throw new IllegalArgumentException("Hash algorithm is blank");
        }
        return valueOf(value.trim().toUpperCase(java.util.Locale.ROOT));
    }
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/security/HashAlgorithm.kt
package ru.privatenull.pnauth.security

import java.util.Locale

enum class HashAlgorithm {
    PBKDF2,
    BCRYPT,
    ARGON2;

    companion object {
        @JvmStatic
        fun parse(value: String?): HashAlgorithm {
            require(!value.isNullOrBlank()) { "Hash algorithm is blank" }
            return valueOf(value.trim().uppercase(Locale.ROOT))
        }
    }
}
